libcoap/src/context.rs

```diff
@@ -406,6 +410,32 @@ impl CoapContext<'_> {
         self.add_endpoint(addr, coap_proto_t_COAP_PROTO_DTLS)
     }
 
+    // TODO: Possibly wrap in feature?
+    /// Joins a multicastgroup
+    /// # Safety
+    /// The libcoap function does not modify/delete any variables here.
+    /// NULL for the interface is handled internally in the libcoap function, therefore safe.
+    pub fn join_mcast_group_intf(
+        &mut self,
+        groupname: &str,
+        ifname: Option<&str>,
+    ) -> Result<(), MulticastGroupJoinError> {
+        let inner_ref = self.inner.borrow();
+        let mcast_groupname = CString::new(groupname).expect("CString::new failed");
+        let mcast_ifname = ifname.and_then(|if_in| Some(CString::new(if_in).expect("CString::new failed")));
+        let mcast_ifname_ptr = match mcast_ifname {
+            Some(if_in) => if_in.as_ptr(),
+            None => std::ptr::null(),
+        };
+        unsafe {
+            let ret = coap_join_mcast_group_intf(inner_ref.raw_context, mcast_groupname.as_ptr(), mcast_ifname_ptr);
```

Comment:
We don't need to create an owned CString here if we allow passing a borrowed CStr ([C-CALLER-CONTROL](https://rust-lang.github.io/api-guidelines/flexibility.html#caller-decides-where-to-copy-and-place-data-c-caller-control)).
```suggestion
        groupname: impl AsRef<CStr>,
        ifname: Option<&CStr>,
    ) -> Result<(), MulticastGroupJoinError> {
        let inner_ref = self.inner.borrow();
        let mcast_groupname = groupname.as_ref().as_ptr();
        let mcast_ifname = ifname.map(|v| v.as_ptr()).unwrap_or(std::ptr::null());
        unsafe {
            let ret = coap_join_mcast_group_intf(inner_ref.raw_context, mcast_groupname, mcast_ifname);
```